### PR TITLE
Robot scene rhino8

### PR DESCRIPTION
**Pull request by chenkasirer**
* Fixed some imports that were no longer working in Rhino8 which caused the `RobotModelObject` to not get registered in context `Rhino`.
* Also upgraded to `compas_invocations2` cause it's almost 2025 ;)
* Replaced other linting tools with `ruff` and the configuration that goes with it.

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [x] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: requirements-dev.txt
Comment (yck011522) [inline requirements-dev.txt:1]: I don't know enough to review these things. What is the difference from compas_invocation2 from the original one? Does these new tools changes the coding style or is simply a drop-in replacement of the old pep8 tools?

Just so I know if I should do the same for compas_fab...

**Review by yck011522 — APPROVED** [review]
I don't know

File: requirements-dev.txt
Comment (chenkasirer) [reply requirements-dev.txt:1]: Yes. `ruff` is pretty much a drop-in for all these other linting and formatting tools we were previously using. 

It's part of a bigger effort to reduce our dependencies and consolidate all these different configuration files (`setup.py`, `setup.conf`, `.bumpversion.cfg` etc.) into a single `pyproject.toml` file. Take a look at core for example.

**Comment by gonzalocasas** [comment]
late to the party, but 🚀 !! Awesome, thanks!

**Pull request merged**